test/Sprites/ImageHeader.test.cpp

```diff
@@ -0,0 +1,71 @@
+#include "../src/Sprites/ImageHeader.h"
+#include <gtest/gtest.h>
+#include <stdexcept>
+#include <cstdint>
+
+TEST(ImageHeader, Create)
+{
+	const int32_t width = 10;
+	const int32_t height = 10;
+	const uint16_t bitCount = 1;
+
+	// Improper Bit Count throws
+	EXPECT_THROW(ImageHeader::Create(width, height, 3), std::runtime_error);
+
+	ImageHeader imageHeader;
+	EXPECT_NO_THROW(imageHeader = ImageHeader::Create(width, height, bitCount));
+
+	EXPECT_EQ(sizeof(ImageHeader), imageHeader.headerSize);
+	EXPECT_EQ(width, imageHeader.width);
+	EXPECT_EQ(height, imageHeader.height);
+	EXPECT_EQ(ImageHeader::defaultPlanes, imageHeader.defaultPlanes);
+	EXPECT_EQ(bitCount, imageHeader.bitCount);
+	EXPECT_EQ(ImageHeader::defaultCompression, imageHeader.compression);
+	EXPECT_EQ(ImageHeader::defaultImageSize, imageHeader.defaultImageSize);
+	EXPECT_EQ(ImageHeader::defaultXResolution, imageHeader.defaultXResolution);
+	EXPECT_EQ(ImageHeader::defaultYResolution, imageHeader.defaultYResolution);
+	EXPECT_EQ(ImageHeader::defaultUsedColorMapEntries, imageHeader.defaultUsedColorMapEntries);
+	EXPECT_EQ(ImageHeader::defaultImportantColorCount, imageHeader.defaultImportantColorCount);
+}
+
+TEST(ImageHeader, IsBitCount)
+{
+	for (auto bitCount : ImageHeader::validBitCounts) {
+		EXPECT_TRUE(ImageHeader::IsBitCount(bitCount));
+	}
+
+	EXPECT_FALSE(ImageHeader::IsBitCount(21));
+}
+
+TEST(ImageHeader, IsIndexedBitCount)
+{
+	for (auto bitCount : ImageHeader::indexedBitCounts) {
+		EXPECT_TRUE(ImageHeader::IsIndexedBitCount(bitCount));
+
+		// Ensure Indexed Bit Count is also a valid general BitCount
+		EXPECT_TRUE(ImageHeader::IsBitCount(bitCount));
+	}
+
+	EXPECT_FALSE(ImageHeader::IsIndexedBitCount(3));
```

Comment:
I think you are recommending that any bit depth less than 8 should pass, even if it is actually an invalid bit depth? My plan was to only allow 1, 4, and 8 to pass as other values would form an invalid bitmap.

Reply:
I'm not sure a bit depth of 3 is necessarily invalid, just not supported by our code. Windows code seems pretty generic in terms of bit depths it supports, and even the ordering of color components. If an image with a bit depth of 3 is produced, it would be palettized. According to what I've read, any bit depth <= 8 requires a palette. As such, I think `IsIndexedBitCount` should return `true`.

The `IsSupported` method can of course return `false`.

Suppose some day we want to support 3 bit color images, which would have a mandatory palette with up to 8 entries. That would mean `IsSupported` would then return `true`. Currently the value of `IsIndexedBitCount` would also change from `false` to `true`, with no change in bit depth. This doesn't seem right. I don't think `IsIndexedBitCount` should depend on `IsSupported` at all. The palettized nature of an image doesn't depend on whether or not we support using that format.

Side note: I was doing some reading and found true color bitmaps can also contain an optional palette for the purpose of optimizing display on low color devices. The palette is mandatory when bit depth <=8, and optional for larger bit depths.

Reply:
I'd like to punt this to an issue and merge. I'd like to resolve some questions I have about how pixels are passed in for less than 8 bit count images before making changes here.